lib/system.nim

```diff
@@ -4198,6 +4198,12 @@ type
   NimNode* {.magic: "PNimrodNode".} = ref NimNodeObj
     ## Represents a Nim AST node. Macros operate on this type.
 
+proc lenVarargsImpl(x: NimNode): NimNode {.magic: "LengthOpenArray", noSideEffect.}
```

Comment:
Why isn't this simply:

```nim

proc lenVarags(x: varargs): int {.magic: "LengthOpenArray", noSideEffect.}
```

Reply:
feel free to try, it doesn't work, nor does these:
```nim
proc lenVarargs*(x: varargs): int {.magic: "LengthOpenArray", noSideEffect.}
../../lib/system.nim(296, 11) Error: invalid expression: [T]

proc lenVarargs*[T](x: varargs[T]): int {.magic: "LengthOpenArray", noSideEffect.}
../../lib/system.nim(4207, 20) Error: invalid type: 'untyped' in this context: 'proc (x: varargs[untyped]): int{.noSideEffect.}' for proc

proc lenVarargs*(x: varargs[untyped]): int {.magic: "LengthOpenArray", noSideEffect.}
tlenvarargs.nim(35, 10) Error: cannot generate VM code for (1, 2)
```

Reply:
So ... fix the compiler that it works?

Reply:
didn't seem easy without lots of surgery because of way varargs are treated, and might open up new bugs; I've instead pushed a commit that moves `lenVarargsImpl` inside `lenVarargs`